File: src/features/material-archive/contracts/material-api-dto.ts
export type MaterialChangeTypeApiDTO = 'MANUAL' | 'ECO' | 'ECN'
export type MaterialStatusApiDTO = 'Active' | 'Inactive' | 'Archived'

export interface MaterialDimensionsApiDTO {
  length: number
  width: number
  height: number
  unit: string
}

export interface MaterialApiDTO {
  id: string
  code: string
  name: string
  category: string
  spec?: string
  internalDimensions?: MaterialDimensionsApiDTO | null
  externalDimensions?: MaterialDimensionsApiDTO | null
  uom: string
  minStock: number
  costPrice?: number
  supplierId?: string
  description?: string
  images?: string[]
  status: MaterialStatusApiDTO
  revisionNo?: string
  effectiveFrom?: string | null
  effectiveTo?: string | null
  changeType?: MaterialChangeTypeApiDTO
  changeOrderNo?: string
  siteCode?: string
  isDefaultSite?: boolean
  createdAt?: string
  updatedAt?: string
  version?: number
}

export interface MaterialOptionApiDTO {
  id: string
  code: string
  name: string
  spec?: string
  uom?: string
  category?: string
  status?: MaterialStatusApiDTO
  costPrice?: number
}

export interface MaterialOptionsResponseApiDTO {
  items: MaterialOptionApiDTO[]
  version: string
}

export interface MaterialListPageApiDTO {
  items: MaterialApiDTO[]
  total: number
  page: number
  pageSize: number
  version: string
}

export interface SaveMaterialApiDTO {
  id?: string
  code: string
  name: string
  category: string
  spec?: string
  internalDimensions?: MaterialDimensionsApiDTO
  externalDimensions?: MaterialDimensionsApiDTO
  uom: string
  minStock: number
  costPrice?: number
  supplierId?: string
  description?: string
  images?: string[]
  status: MaterialStatusApiDTO
  revisionNo?: string
  effectiveFrom?: string | null
  effectiveTo?: string | null
  changeType?: MaterialChangeTypeApiDTO
  changeOrderNo?: string
  siteCode?: string
  isDefaultSite?: boolean
  version?: number
}

export interface BulkSyncMaterialsApiDTO {
  materials: SaveMaterialApiDTO[]
  globalVersion?: number
}

File: server/handlers/material_api_dto.go
package handlers

type MaterialApiDTO struct {
	ID                 string  `json:"id"`
	Code               string  `json:"code"`
	Name               string  `json:"name"`
	Category           string  `json:"category"`
	Spec               string  `json:"spec"`
	InternalDimensions any     `json:"internalDimensions"`
	ExternalDimensions any     `json:"externalDimensions"`
	UOM                string  `json:"uom"`
	MinStock           float64 `json:"minStock"`
	CostPrice          float64 `json:"costPrice"`
	SupplierID         string  `json:"supplierId"`
	Description        string  `json:"description"`
	Images             any     `json:"images"`
	Status             string  `json:"status"`
	// --- MasterDataControl 嵌套命名空间（唯一输出格式） ---
	MasterDataControl  *MasterDataControlDTO `json:"masterDataControl,omitempty"`
	CreatedAt          any     `json:"createdAt"`
	UpdatedAt          any     `json:"updatedAt"`
	Version            int     `json:"version"`
}

type MaterialListPageApiDTO struct {
	Items    []MaterialApiDTO `json:"items"`
	Total    int64            `json:"total"`
	Page     int              `json:"page"`
	PageSize int              `json:"pageSize"`
	Version  string           `json:"version"`
}

type MaterialOptionApiDTO struct {
	ID        string  `json:"id"`
	Code      string  `json:"code"`
	Name      string  `json:"name"`
	Spec      string  `json:"spec"`
	UOM       string  `json:"uom"`
	Category  string  `json:"category"`
	Status    string  `json:"status"`
	CostPrice float64 `json:"costPrice"`
}

type MaterialOptionsApiDTO struct {
	Items   []MaterialOptionApiDTO `json:"items"`
	Version string                 `json:"version"`
}
